research-software-directory/RSD-as-a-service
# feat: implement community settings page

# Community settings

Closes #1184

Changes proposed in this pull request:
* Create community settings where maintainers can edit community settings or invite new maintainers  
* Refactor maintainers sections: created maintainers shared components and methods which are used by software, project, organisations and communities.
* Additional unit tests for maintainer sections are created

How to test:
* `make start` to build and generate test data
* login as rsd-admin and navigate naar een community, or create new community from admin section
* navigate to community settings and change the values, add primary maintainer
* navigate to maintainers section, create invite links and use them with another user. Confirm the maintainer is added to the list. Confirm that regular maintainer does not see admin section of general settings.
* removing maintainers: confirm that only rsd-admin can remove primary maintainer. The regular maintainer can remove itself or other maintainers but it cannot remove the last "regular" maintainer.
* change the values in about page or remove about page completely by deleting the content
* double check maintainers sections for: software, projects and organisations too.

## Community settings
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/9944ba24-305c-4fb6-853b-ed543f78cd97)

## Community mantainers
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/d5bdfd9a-07b7-40ca-bc6b-6fab004f8907)

## Community about page
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/2f40a032-6514-4864-9cfc-74984e7548ff)

PR Checklist:
